src/ixdat/db.py

```diff
@@ -197,33 +230,63 @@ def set_backend(self, backend):
         """Backends set obj.backend here after loading/saving a Saveable obj"""
         self._backend = backend
 
-    def get_main_dict(self):
-        """Return dict: serializition only of the row of the object's main table"""
+    def get_main_dict(self, exclude=None):
+        """Return dict: serializition only of the row of the object's main table
+
+        Args:
+            exclude (list): List of attribute names to leave out of the dict
+        """
+        exclude = exclude or []
         if self.column_attrs is None:
             raise DataBaseError(
                 f"{self} can't be serialized because the class "
                 f"{self.__class__.__name__} hasn't defined column_attrs"
             )
-        self_as_dict = {attr: getattr(self, attr) for attr in self.column_attrs}
+        self_as_dict = {  # FIXME: probably better as loop, fix with table definitions.
```

Comment:
Agreed, this is too complex to have the best readability  as a comprehension.

Reply:
Next PR.